ansible/roles/common/templates/boot.sh

```diff
@@ -21,6 +21,7 @@ cd /opt/common
 if [ ! -f first_boot_done ] && [ ! -f first_boot_failed ]; then
   echo "Starting first boot."
   if ./initialize.sh ; then
+    /opt/common/libexec/invoke-hook first-boot
```

Comment:
waaaait.

So the way I see it, we should have hooks:

- `first-boot`, that right now only starts the DHCP server and sets up the initial lan. In the future, we'll also serve the `welcome` pages from this hook
- future hook: `after-welcome`, that runs the things that change the `liquid_domain` (like the matrix configs), sets up the encrypted partitions and so on
- not currently a hook: `initialize`, that makes sure all apps are set up and running

Can we run the `first-book` hook before the `initialize.sh` script? Even if that means we duplicate calls used to set up the LAN, instead of putting some configuration in `liquid-core`. We won't have `liquid-core` set up at that first boot, will we?

Reply:
For the benefit of future code archaeologists: @gabriel-v makes some valid points, and we agreed to revisit this issue after we ship a useful odroid c2 image. The current configuration/boot/initialize story is crazyland.